onnxruntime/test/platform/threadpool_test.cc

```diff
@@ -410,4 +410,92 @@ TEST(ThreadPoolTest, TestStackSize) {
+    ASSERT_TRUE(num_non_default_affinities == num_threads);
+  }
+#endif
+}
+
+TEST(ThreadPoolTest, TestAffinityStringMisshaped) {
+  std::vector<uint64_t> default_affinities;
+  int default_threadpool_size = static_cast<int>(Env::Default().GetDefaultThreadpoolSetting(default_affinities));
+  ASSERT_TRUE(default_threadpool_size > 0);
+#ifdef _WIN32
+  // test only one machines has more than 8 logical processors
+  if (default_threadpool_size >= 4) {
+    std::vector<uint64_t> affinities;
+    // cases - test processor id that's out of boundary 
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig("0", affinities));
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig(std::to_string(default_threadpool_size*2+1), affinities));
+    // cases - test empty strings
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig(",", affinities));
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig(",1", affinities));
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig(";", affinities));
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig(";1", affinities));
+    // cases - test invalid chars
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig("2,a,1", affinities));
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig("*4,3,)", affinities));
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig("3,5,1;^.2", affinities));
+    // cases - test comma wrongly coupled with hyphen
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig("1,-5", affinities));
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig("2,3,-", affinities));
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig("3,5-1;6,8", affinities));
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig("-,-", affinities));
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig("1-2;3-4,", affinities));
+    // cases - test hyphen of invalid intervals
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig("6-2", affinities));
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig("-2", affinities));
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig("1-2;4-3", affinities));
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig("1--2;3-4", affinities));
+    ASSERT_TRUE(0 == Env::Default().ReadThreadAffinityConfig("1-3-4", affinities));
+  }
+#endif
+}
+
+#ifdef _WIN32
+TEST(ThreadPoolTest, TestNuma_2_Group_64_core_128_Processor) {
+  auto& env = Env::Default();
+  auto system_info_snap = env.GetSystemInfo();
+  int32_t logical_processors_per_core = 2;
+  int32_t cores_per_group = 32;
+  int32_t num_cores = 64;
+  env.SetSystemInfo({logical_processors_per_core, cores_per_group, num_cores});
+  std::vector<uint64_t> default_affinities;
+  int default_threadpool_size = static_cast<int>(Env::Default().GetDefaultThreadpoolSetting(default_affinities));
+  ASSERT_TRUE(default_threadpool_size == 64);
```

Comment:
1. Is this enough to give us confidence? What about the affinities vector returned?
2. Is there a test that combines a custom affinity string along with this test?

Reply:
The string parsing code is the worry. Plus, we have a new C API, that needs to be covered.

Reply:
A set of new test cases have been added addressed varied group + core cases.
Besides, a case covering the API also added in onnxruntime_global_thread_pools_test.